iterative/azure/provider.go

```diff
@@ -360,6 +364,27 @@ func getVMClient(subscriptionID string) (compute.VirtualMachinesClient, error) {
 	return client, err
 }
 
+// getUserAssignedIdentityMap generates a map of user-assigned identity values from a comma-separated
+// list of ARM resource ids.
+func getUserAssignedIdentityMap(identitiesRaw string) map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue {
```

Comment:
This code is very similar to 

```
identities := strings.Split(ps.Identifier, ",")
	identityMap := map[string]*compute.VirtualMachineScaleSetIdentityUserAssignedIdentitiesValue{}
	for _, identity := range identities {
		identity = strings.TrimSpace(identity)
		if identity == "" {
			continue
		}
		if !validateARMID.MatchString(identity) {
			return fmt.Errorf("invalid user-assigned identity id: %q", identity)
		}
		identityMap[identity] = &compute.VirtualMachineScaleSetIdentityUserAssignedIdentitiesValue{}
	}
	if len(identityMap) == 0 {
		ps.Resource = nil
		return nil
	}
	ps.Resource = &compute.VirtualMachineScaleSetIdentity{
		UserAssignedIdentities: identityMap,
```

are they different in scope? Could them be using a common function?

Reply:
They are pretty much the same thing, but I've not yet found a good place to put common azure-specific functionality. Also, there's the subtle difference that in one place we're creating a map of `*compute.VirtualMachineIdentityUserAssignedIdentitiesValue` and in the other - a map of `*compute.VirtualMachineScaleSetIdentityUserAssignedIdentitiesValue`. 

DRY is a nice principle in theory, in practice it comes with its own cost.

Reply:
Im not that picky but fi there is an easy way to just call a generic map creator would be awesome.